app/models/spree/image_decorator.rb

```diff
@@ -14,7 +14,7 @@ def update_mailchimp_product
       end
 
       def find_product
-        Spree::Variant.find(self.viewable_id).product.id
+        Spree::Variant.unscoped.find(viewable_id).product_id
```

Comment:
The method is called `find_product` but it returns a product ID. Maybe we could change the query to something like:

```
Spree::Product.joins(:variants).find_by(spree_variants: { id: viewable_id })
```